sympy/printing/str.py

```diff
@@ -74,13 +74,16 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~(%s)' %(expr.args[0])
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return '%s' %(' & '.join([self.parenthesize(arg, 38)
```

Comment:
I cannot use `PRECEDENCE["And"]` because `PRECEDENCE["And"]` < `PRECEDENCE["RELATIONAL"]`.

However in python, precedence of `&` is greater than precedence of relational operators.
This means Expressions like : 
```python
And(x<=y,y<=z)
```
will be printed as : 
```python
x <= y & y <= z
```
However, it should be printed as : 
```python
(x <= y) & (y <= z)
```